userspace/libsinsp/filterchecks.cpp

```diff
@@ -90,6 +90,13 @@ int32_t gmt2local(time_t t)
 	return dt;
 }
 
+static inline bool str_match_start(const std::string& val, size_t len, const char* m)
+{
+	return val.compare(0, len, m) == 0;
```

Comment:
Similar to one of my previous comments, though a little harder to implement since `val` is used in multiple places after the comparison, maybe we could keep it as a raw char* for comparisons and turn it into a string only when needed, WDYT?

Reply:
leaving this for the next iteration